api/rc3.py

```diff
@@ -89,15 +89,18 @@
 	methods=["GET", "OPTIONS"]
 )
 async def devices(rq, host, port):
-	""" Device list """
+	""" Device list 
+	url args can be ?domain=reg1&family=reg2&member=reg3&wildcard=sys/tg_test&range=0-25
+	if pattern is present the other filters are ignored
+	"""
 	devs = []
-	domains = db.get_device_domain("*")
-	for d in domains:
-		families = db.get_device_family("%s/*" % d)
-		for f in families:
-			members = db.get_device_member("%s/%s/*" % (d, f))
-			for m in members:
-				devs.append('/'.join((d, f, m)))
+
+	dev_range = rq.args.pop("range", None)
+	filters = rq.args
+	db_proxy = await getDeviceProxy('sys/database/2')
```

Comment:
'sys/database/2' is not always the database
you can use db (PyTango.Database()) which give you a proxy on the default database. You may not profit of the async method but the previous code did not use it.

Reply:
`db` does not give a method for getting the devices list. is there any way of guessing the db device name from the `db`?

Reply:
`db.get_name()`